scripts/install/production/roles/bmi/tasks/main.yml

```diff
@@ -0,0 +1,69 @@
+---
+#This role performs the steps to install BMI
+
+- name: Create log and pxelinux directories
+  file:
+    state: directory
+    group: "{{ lookup('env', 'USER') }}"
+    owner: "{{ lookup('env', 'USER') }}"
+    path: "{{ item }}"
+  become: true
+  with_items:
+    - "/var/log/bmi/logs"
+    - "/etc/bmi/pxelinux.cfg"
+
+- name: Copy config, ipxe and mac templates to the correct directories
+  copy:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    owner: "{{ lookup('env', 'USER') }}" 
+    group: "{{ lookup('env', 'USER') }}" 
+    mode: 0664
+  become: true
+  with_items:
+    - { src: '../../../bmi_config.cfg', dest: '/etc/bmi/bmiconfig.cfg' }
+    - { src: '../../../ims/ipxe.temp', dest: '/etc/bmi/ipxe_example.temp' }
+    - { src: '../../../ims/mac.temp', dest: '/etc/bmi/pxelinux.cfg/mac_example.temp' }
+
+- name: Install setup.py
+  command: "python setup.py install"
+  args:
+    chdir: "{{playbook_dir}}/../../.."
+  become: true
+
+- name: Install cephlibs
+  pip:
```

Comment:
Why are you using pip for this ? Packages should be coming from yum repos on centos you can get it by
yum -y install http://download.ceph.com/rpm-luminous/el7/noarch/ceph-release-1-1.el7.noarch.rpm
and then yum install whatever you need from ceph

On rhel there is some code in the dev scripts using rhcs 1.2, if you change that to 2x you will get the up to date packages or you can use upstream the same as above on CentOS

Does pip install latest dev code for ceph? Latest dev != production

Reply:
As far as I can tell, that package actually isn't a ceph package, it is a package created by a 3rd party that provides rados and rbd python bindings to connect BMI to an exisiting ceph cluster.  I also believe the only way to install it is through pip.

Reply:
https://pypi.python.org/pypi/python-cephlibs/ is what you guys use? hmmm... two years old code marked as deprecated? you should probably use the official rados/rbd bindings (whose source code is here https://github.com/ceph/ceph/tree/master/src/pybind and listed on pypi simply as "rados" and "rbd"). I'll make an issue about it. (But for now, what you're doing is fine)